### ISSUE TITLE
Double loop on host when including a role with a with_items loop

### ISSUE THREAD

**Issue by Raoul555** [ansible/ansible#31422]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
- play
- role
- loop
- with_items

##### ANSIBLE VERSION
ansible 2.2.2.0
  config file = 
  configured module search path = Default w/o overrides

##### CONFIGURATION
None

##### OS / ENVIRONMENT
Ansile OS: debian 9

##### SUMMARY
When using calling a role from a play with a with_items loop for a var, ansible is making a double loop on each hosts

##### STEPS TO REPRODUCE
ansible-playbook test.yml

test.yml:
```
plays$ cat test.yml
---
- name: Test
  hosts: all
  gather_facts: no

  tasks:

    - name: Enter role
      include_role:
          name: sib
      vars:
          foo: "{{ item }}"
      with_items:
          - bar
```
sib/tasks/main.yml
```
plays$ cat roles/sib/tasks/main.yml 
---

- debug:
    var: foo
```
##### EXPECTED RESULTS
```
plays$ ansible-playbook -i ~/dev/project/base/ansible/hosts test.yml
 
PLAY [Test] ********************************************************************

TASK [Enter role] **************************************************************

TASK [sib : debug] *************************************************************
ok: [fe1] => {
    "foo": "bar"
}
ok: [fe2] => {
    "foo": "bar"
}
ok: [be1] => {
    "foo": "bar"
}
ok: [be2] => {
    "foo": "bar"
}
```
##### ACTUAL RESULTS
```
plays$ ansible-playbook -i ~/dev/project/base/ansible/hosts test.yml
 
PLAY [Test] ********************************************************************

TASK [Enter role] **************************************************************

TASK [sib : debug] *************************************************************
ok: [fe1] => {
    "foo": "bar"
}
ok: [fe2] => {
    "foo": "bar"
}
ok: [be1] => {
    "foo": "bar"
}
ok: [be2] => {
    "foo": "bar"
}

TASK [sib : debug] *************************************************************
ok: [be1] => {
    "foo": "bar"
}
ok: [fe2] => {
    "foo": "bar"
}
ok: [fe1] => {
    "foo": "bar"
}
ok: [be2] => {
    "foo": "bar"
}

TASK [sib : debug] *************************************************************
ok: [fe1] => {
    "foo": "bar"
}
ok: [fe2] => {
    "foo": "bar"
}
ok: [be1] => {
    "foo": "bar"
}
ok: [be2] => {
    "foo": "bar"
}

TASK [sib : debug] *************************************************************
ok: [fe1] => {
    "foo": "bar"
}
ok: [fe2] => {
    "foo": "bar"
}
ok: [be1] => {
    "foo": "bar"
}
ok: [be2] => {
    "foo": "bar"
}

PLAY RECAP *********************************************************************
be1                        : ok=4    changed=0    unreachable=0    failed=0   
be2                        : ok=4    changed=0    unreachable=0    failed=0   
fe1                        : ok=4    changed=0    unreachable=0    failed=0   
fe2                        : ok=4    changed=0    unreachable=0    failed=0   
```

**Comment by techraf** [comment]
Likely a duplicate of #18748

**Comment by Raoul555** [comment]
So, it's a bug. But it does not seem to have been fixes in 2.2

I've still have one question. My use case looks to me very simple:
- I've wrote a role to install a package
- I've defined in my group_vars the list of packages to be installed
- I've wrote a play in which I simply call my role (include_role) looping one my packages list (with_items)

How come I hit this bug on this simple case? Is there a more ansile way of writting this? How does others do?

**Comment by sivel** [comment]
Based on the above referenced duplicate, this issue has been resolved on the 2.4 release. Please upgrade to 2.4 and verify.

needs_info

**Comment by Raoul555** [comment]
Will there be a fix for 2.2?

**Comment by mkrizek** [comment]
@Raoul555 no, there won't be another 2.2 release, unfortunately.

If you have further questions please stop by IRC or the mailing list:

- IRC: #ansible on irc.freenode.net
- mailing list: https://groups.google.com/forum/#!forum/ansible-project
